components/Tag/Tag.jsx

```diff
@@ -0,0 +1,96 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+function getClass(isHover) {
+  return classNames({
+    inline: true,
+    relative: true,
+    'c-tag': true,
+    'is-hover': isHover,
+  });
+}
+
+function getButtonClass(isHover, isProcessing) {
+  return classNames({
+    'c-tag--button': true,
+    'btn-teal': isHover,
+    'is-hover': isHover,
+    'btn-gray': !isHover,
+    'is-processing': isProcessing,
+  });
+}
+
+function getLinkClass(isRemoveHover) {
+  return classNames({
+    'c-tag--remove': true,
+    'icon--center': true,
+    'icon-x-white': true,
+    'icon--xxsmall': true,
+    'btn-red': isRemoveHover,
+    'btn-teal': !isRemoveHover,
+  });
+}
+
+function truncate(str, maxLength) {
+  return str.length > maxLength ? str.slice(0, maxLength).concat('...') : str;
+}
```

Comment:
Also - I have a feeling we'll probably use this `truncate` function elsewhere, so maybe we could move this to the utils file.